scrapy/spiders/__init__.py

```diff
@@ -34,12 +52,14 @@ class Spider(object_ref):
     def __init__(self, name: Optional[str] = None, **kwargs: Any):
         if name is not None:
             self.name = name
-        elif not getattr(self, "name", None):
-            raise ValueError(f"{type(self).__name__} must have a name")
```

Comment:
It looks like spider.name is going to raise an AttributeError for nameless spiders. Won't it break spider.name usages found by @GeorgeA92 in https://github.com/scrapy/scrapy/issues/1157#issuecomment-1816782204?